geode-core/src/main/java/org/apache/geode/management/internal/cli/functions/CreateDefinedIndexesFunction.java

```diff
@@ -55,8 +69,33 @@ public void execute(FunctionContext context) {
           queryService.defineIndex(indexName, indexedExpression, regionPath);
         }
       }
-      indexes = queryService.createDefinedIndexes();
-      context.getResultSender().lastResult(new CliFunctionResult(memberId));
+
+      List<Index> indexes = queryService.createDefinedIndexes();
+
+      // One XmlEntity per region.
+      ResultSender sender = context.getResultSender();
+      List<String> processedRegions = new ArrayList<>();
+      List<CliFunctionResult> functionResults = new ArrayList<>();
+
+      for (Index index : indexes) {
+        String regionName = index.getRegion().getName();
+
+        if (!processedRegions.contains(regionName)) {
+          XmlEntity xmlEntity = createXmlEntity(regionName);
+          functionResults.add(new CliFunctionResult(memberId, xmlEntity));
+          processedRegions.add(regionName);
+        }
+      }
+
+      int resultsSize = functionResults.size();
+
+      if (resultsSize > 0) {
```

Comment:
still, this outer check is unnecessary

Reply:
Thank you so much. The tests are great. just some minor suggestions.

Reply:
For this particular case I'm accessing the elements by index using `functionResults.get(i)` and `functionResults.get(resultsSize - 1)` within a `for (int i = 0; i < resultsSize - 1; i++)` loop, so I need the check to avoid any `IndexOutOfBoundsException`.

Reply:
so, if your resultSize == 0, you don't even call a sender.lastResult? I thought you should always do a lastResult call.

Reply:
You're right, I should always call `sender.lastResult()` no matter what... I'll have another look at the logic and submit the changes altogether. Thanks for reviewing :-).